components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -530,13 +530,15 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                 LOG.debug("Retrieved the User Information:" + jwtAttributeMap);
             }
 
-            String nonce = (String) jwtAttributeMap.get(NONCE);
-            if (nonce == null) {
-                LOG.debug("OIDC provider does not support nonce claim in id_token.");
-            }
-            if (nonce != null && !nonce.equals(context.getProperty(NONCE))) {
-                throw new AuthenticationFailedException(ErrorMessages.NONCE_MISMATCH.getCode(),
-                        ErrorMessages.NONCE_MISMATCH.getMessage());
+            if (context.getProperty(NONCE) != null) {
```

Comment:
Saw that we are setting property to the context from the constant org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.Claim.NONCE which is simply "nonce". 

Can we introduce a new constant for this and make it specific for the authenticator. something like "OIDC_FEDERATION_NONCE" (you can come up with a better name). The reason is the authentication context is used in various places and we are using a general property map to store this, so if any other component decided that its also going to store a nonce then we will start seeing erroneous results here. So its good to have a unique property name that is unlikely to be used by some other component.

Reply:
+1. Thanks for the suggestion and will send an improvement

Reply:
Fixed by - https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/142/commits/1e00ef0a95ea6894616780d0712fff8d89475f7a